WordPress-Core/ruleset.xml

```diff
@@ -45,10 +45,18 @@
 	<!-- Covers rule: Use real tabs and not spaces. -->
 	<rule ref="Generic.WhiteSpace.DisallowSpaceIndent"/>
 
-	<!-- Covers rule: Note the comma after the last array item: this is recommended. -->
+	<!-- Generic array layout check. -->
 	<rule ref="WordPress.Arrays.ArrayDeclaration">
 		<exclude name="WordPress.Arrays.ArrayDeclaration.SingleLineNotAllowed"/>
+		<exclude name="WordPress.Arrays.ArrayDeclaration.CommaAfterLast"/>
+		<exclude name="WordPress.Arrays.ArrayDeclaration.NoSpaceAfterComma"/>
+		<exclude name="WordPress.Arrays.ArrayDeclaration.SpaceAfterComma"/>
+		<exclude name="WordPress.Arrays.ArrayDeclaration.SpaceBeforeComma"/>
 	</rule>
+
+	<!-- Covers rule: Note the comma after the last array item: this is recommended. -->
+	<rule ref="WordPress.Arrays.CommaAfterArrayItem"/>
```

Comment:
> Just so I'm clear - this new sniff replaces the 4 excluded checks above

Just to be clear :heart_eyes:  : this sniff replaces the 4 excluded checks above (= single line array comma checks as we defer to the upstream method for the single line array analysis) :heavy_plus_sign:  the newly commented out code in the `ArrayDeclaration` sniff (= multi-line array comma checks as we overload that method for the multi-line array analysis ).

When combined - as I did now -, the code for all those checks together contained quite a lot of duplication and some redundancy, which has all been removed for this sniff.
I've kept the error message texts and the sniff codes close to the original code though.

> which will later be removed?

At some point in the future, I will pull this sniff + a few more of the array sniffs I've created (and one or two more which I probably will create still), upstream, https://github.com/squizlabs/PHP_CodeSniffer/issues/582 will be fixed and the upstream `Squiz.Arrays.ArrayDeclaration` sniff will effectively be reduced to the minimal code which is Squiz specific, everything else being individual sniffs in `Generic`.
At that point, we can remove both our versions of these new sniffs, as well as our copy of the `ArrayDeclaration` sniff and defer to using the newly introduced upstream versions of these sniffs.

Until that time, we will need the exclusions. As for the commented out code in the sniff copy, that will remain in the WP version of the `ArrayDeclation` sniff to facilitate easier compare with potential upstream changes to that sniff until we remove the sniff completely.


Does that help ?